lib/lib-utils/src/main/java/org/sagebionetworks/util/json/JavaJSONUtil.java

```diff
@@ -0,0 +1,192 @@
+	}
+
+	@SuppressWarnings("unchecked")
+	static <F, J> Optional<JSONObject> writeToJSON(List<Translator<?, ?>> translators, Object object) {
+		ValidateArgument.required(translators, "translators");
+		ValidateArgument.required(object, "object");
+
+		JSONObject json = new JSONObject();
+		boolean wasWritten = false;
+		Class<? extends Object> clazz = object.getClass();
+		for (Field field : clazz.getDeclaredFields()) {
+			if (!Modifier.isStatic(field.getModifiers())) {
+				try {
+					field.setAccessible(true);
+					Object value = field.get(object);
+					if (value != null) {
+						Translator<F, J> transaltor = findTranslator(translators, field.getType());
+						json.put(field.getName(), transaltor.translateFromJavaToJSON((F) value));
+						wasWritten = true;
+					}
+				} catch (IllegalArgumentException | IllegalAccessException e) {
+					throw new RuntimeException(e);
+				}
+			}
+		}
+		return wasWritten ? Optional.of(json) : Optional.empty();
+	}
+
+	/**
+	 * Helper to find a translator for the provided type.
+	 * 
+	 * @param <F>
+	 * @param <J>
+	 * @param translators The is of possible translator.
+	 * @param type
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	static <F, J> Translator<F, J> findTranslator(List<Translator<?, ?>> translators, Class<?> type) {
+		ValidateArgument.required(translators, "translators");
+		ValidateArgument.required(type, "type");
+
+		Translator<F, J> transaltor = (Translator<F, J>) translators.stream().filter(t -> t.canTranslate(type))
+				.findFirst()
+				.orElseThrow(() -> new IllegalArgumentException("No translator found for: " + type.getName()));
+		return transaltor;
+	}
+
+	@SuppressWarnings("unchecked")
+	static <T, F, J> T readFromJSON(List<Translator<?, ?>> translators, Class<? extends T> clazz,
+			JSONObject jsonObject) {
+		ValidateArgument.required(translators, "translators");
+		ValidateArgument.required(clazz, "type");
+		ValidateArgument.required(jsonObject, "jsonObject");
+
+		try {
+			T newObject = (T) createNewInstance(clazz);
+			for (Field field : clazz.getDeclaredFields()) {
+				if (!Modifier.isStatic(field.getModifiers())) {
+					if (jsonObject.has(field.getName())) {
+						field.setAccessible(true);
+						Translator<F, J> transaltor = findTranslator(translators, field.getType());
+						J jsonValue = (J) JSONType.lookupType(transaltor.getJSONClass()).getFromJSON(field.getName(),
+								jsonObject);
+						field.set(newObject, transaltor.translateFromJSONToJava((Class<F>) field.getType(), jsonValue));
+					}
+				}
+			}
+			return newObject;
+		} catch (IllegalAccessException | IllegalArgumentException e) {
+			throw new IllegalArgumentException(e);
+		}
+	}
+
+	/**
+	 * Create a new Java object from the given class. The calls must provide a zero
+	 * argument constructor.
+	 * 
+	 * @param type
+	 * @return
+	 */
+	public static Object createNewInstance(Class<?> type) {
+		Optional<Constructor<?>> constructor = Arrays.stream(type.getDeclaredConstructors())
```

Comment:
I think you can just try to get the constructor without a parameter dirctly rather than streaming